src/main/java/org/jabref/gui/contentselector/ContentSelectorDialogView.java

```diff
@@ -0,0 +1,125 @@
+
+import org.jabref.gui.BasePanel;
+import org.jabref.gui.DialogService;
+import org.jabref.gui.JabRefFrame;
+import org.jabref.gui.util.BaseDialog;
+import org.jabref.gui.util.ControlHelper;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.metadata.MetaData;
+
+import com.airhacks.afterburner.views.ViewLoader;
+
+public class ContentSelectorDialogView extends BaseDialog<Void> {
+
+    private static final int FIRST_ELEMENT = 0;
+
+    @FXML
+    private Button addFieldNameButton;
+    @FXML
+    private Button removeFieldNameButton;
+    @FXML
+    private Button addKeywordButton;
+    @FXML
+    private Button removeKeywordButton;
+    @FXML
+    private ListView<String> fieldNamesListView;
+    @FXML
+    private ListView<String> keywordsListView;
+    @FXML
+    private ButtonType saveButton;
+
+    private ContentSelectorDialogViewModel viewModel;
+
+    private final BasePanel basePanel;
+    private final DialogService dialogService;
+    private final MetaData metaData;
+
+    public ContentSelectorDialogView(JabRefFrame jabRefFrame) {
+        this.setTitle(Localization.lang("Manage content selectors"));
+        this.getDialogPane().setPrefSize(375, 475);
+
+        this.basePanel = jabRefFrame.getCurrentBasePanel();
+        this.dialogService = jabRefFrame.getDialogService();
+        this.metaData = basePanel.getBibDatabaseContext().getMetaData();
+
+        ViewLoader.view(this)
+                .load()
+                .setAsDialogPane(this);
+
+        ControlHelper.setAction(saveButton, getDialogPane(), event -> saveChangesAndClose());
+    }
+
+    @FXML
+    public void initialize() {
+        viewModel = new ContentSelectorDialogViewModel(metaData, basePanel, dialogService);
+
+        initListView(fieldNamesListView, viewModel::getFieldNamesBackingList, (observable, oldValue, newValue) -> onFieldNameSelected(newValue));
+        initListView(keywordsListView, viewModel::getKeywordsBackingList, (observable, oldValue, newValue) -> onKeywordSelected());
+    }
+
+    @FXML
+    private void addNewFieldName() {
+        viewModel.showInputFieldNameDialog();
+    }
+
+    @FXML
+    private void removeFieldName() {
+        getSelectedFieldName().ifPresent(viewModel::showRemoveFieldNameConfirmationDialog);
+    }
+
+    @FXML
+    private void addNewKeyword() {
+        getSelectedFieldName().ifPresent(viewModel::showInputKeywordDialog);
+    }
+
+    @FXML
+    private void removeKeyword() {
+        Optional<String> fieldName = getSelectedFieldName();
+        Optional<String> keywordToRemove = getSelectedKeyword();
+        if (fieldName.isPresent() && keywordToRemove.isPresent()) {
+            viewModel.showRemoveKeywordConfirmationDialog(fieldName.get(), keywordToRemove.get());
+        }
+    }
+
+    private void initListView(ListView<String> listViewToInit, Supplier<ObservableList<String>> backingList, ChangeListener<String> onSelectedListener) {
+        listViewToInit.setItems(backingList.get());
```

Comment:
Better use bindings, so you can get a better separation between the view and the ViewModel.

myListView.itemsProperty().bind(viewModel.backingListProperty);
Same also applies to selectedItemProperty 
Sth like this          `viewModel.seletedItemProperty().bind(listView.getSelectionModel().selectedItemProperty());`


This also removes the need to have the ChangeListener, because the bindings under the hood have the change listener implemented

Reply:
Fixed 205170c